qubesadmin/tools/qvm_ls.py

```diff
@@ -594,60 +615,115 @@ def get_parser():
         action='store', choices=formats.keys(), default='simple',
         help='preset format')
 
-    parser_formats.add_argument('--fields', '-O', metavar='FIELD,...',
+    parser_format_group.add_argument('--fields', '-O', metavar='FIELD,...',
         action='store',
         help='user specified format (see available columns below)')
 
-
-    parser.add_argument('--tags', nargs='+', metavar='TAG',
-        help='show only VMs having specific tag(s)')
-
-    for pwrstate in DOMAIN_POWER_STATES:
-        parser.add_argument('--{}'.format(pwrstate), action='store_true',
-            help='show {} VMs'.format(pwrstate))
-
-    parser.add_argument('--raw-data', action='store_true',
-        help='Display specify data of specified VMs. Intended for '
-             'bash-parsing.')
-
-    parser.add_argument('--tree', '-t',
+    parser_format.add_argument('--tree', '-t',
         action='store_const', const='tree',
         help='sort domain list as network tree')
 
-    parser.add_argument('--spinner',
-        action='store_true', dest='spinner',
-        help='reenable spinner')
-
-    parser.add_argument('--no-spinner',
-        action='store_false', dest='spinner',
-        help='disable spinner')
+    parser_format.add_argument('--raw-data', action='store_true',
+        help='Display specify data of specified VMs. Intended for '
+             'bash-parsing.')
 
     # shortcuts, compatibility with Qubes 3.2
-    parser.add_argument('--raw-list', action='store_true',
+    parser_format.add_argument('--raw-list', action='store_true',
         help='Same as --raw-data --fields=name')
 
-    parser.add_argument('--disk', '-d',
+    parser_format_group.add_argument('--disk', '-d',
         action='store_const', dest='format', const='disk',
         help='Same as --format=disk')
 
-    parser.add_argument('--network', '-n',
+    parser_format_group.add_argument('--network', '-n',
         action='store_const', dest='format', const='network',
         help='Same as --format=network')
 
-    parser.add_argument('--kernel', '-k',
+    parser_format_group.add_argument('--kernel', '-k',
         action='store_const', dest='format', const='kernel',
         help='Same as --format=kernel')
 
+    parser_format.add_argument('--help-formats', action=_HelpFormatsAction)
+    parser_format.add_argument('--help-columns', action=_HelpColumnsAction)
+
+    parser_filter = parser.add_argument_group(title='filtering options')
+
+    parser_filter.add_argument('--class', nargs='+', metavar='CLASS',
+        dest='klass', action='store',
+        help='show only VMs of specific class(es)')
+
+    parser_filter.add_argument('--label', nargs='+', metavar='LABEL',
+        action='store',
+        help='show only VMs with specific label(s)')
+
+    parser_filter.add_argument('--tags', nargs='+', metavar='TAG',
+        help='show only VMs having specific tag(s)')
+
+    parser_filter.add_argument('--no-tags', nargs='+', metavar='TAG',
```

Comment:
I understand many unix tools has the negating option prefix `--no-something`, but it doesn't mean it will print qubes that have no tags, but qubes that don't have a certain tag. For this reason and because `qvm_ls` already has the `--exclude` option, please change `--no-tags` to `--exclude-tags`.